pep-9999.rst

```diff
@@ -0,0 +1,464 @@
+PEP: 9999
+Title: Deep Immutability in Python
+Author: Matthew Johnson <matjoh@microsoft.com>, Matthew Parkinson <mattpark@microsoft.com>, Sylvan Clebsch <sylvan.clebsch@microsoft.com>, Fridtjof Peer Stoldt <fridtjof.stoldt@it.uu.se>, Tobias Wrigstad <tobias.wrigstad@it.uu.se>
+Sponsor: TBD
+Discussions-To: TBD
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 27-Feb-2025
+Python-Version: 3.12
+Post-History: 
+Resolution:
+
+
+Abstract
+========
+
+This PEP proposes adding a mechanism for deep immutability to
+Python via a new builtin function ``freeze(obj)``. This function
+takes a reference to an object and recursively render the object
+and all objects it references immutable. (This is *deep*
+immutability --- just making the first object immutable is called
+*shallow* immutability.)
+
+Deep immutability will provide stronger guarantees against
+unintended modifications, improving correctness, security, and
+parallel execution safety.
+
+In many respects, this proposal can be thought of as an
```

Comment:
Not clear to me what is being said here. Immortal objects are mutable, they just never get GCed.

Reply:
Good point. I've killed this sentence.